tableaudocumentapi/archivefile.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+import shutil
+import tempfile
+import zipfile
+
+import xml.etree.ElementTree as ET
+
+
+@contextlib.contextmanager
+def temporary_directory(*args, **kwargs):
+    d = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield d
+    finally:
+        shutil.rmtree(d)
+
+
+def find_file_in_zip(zip, ext):
+    for filename in zip.namelist():
+        if os.path.splitext(filename)[-1].lower() == ext[:-1]:
```

Comment:
I came up with this, and it works, should protect against a random XML file and bad file extensions:

`def find_file_in_zip(zip, ext):
    for filename in zip.namelist():
        try:
            ET.parse(zip.open(filename)).getroot().tag in ('workbook', 'datasource')
            return filename
        except ET.ParseError:
            # that ain't it
            pass`

Not sure how much better than checking the ext it is though